packages/extension-ui/src/Popup/Accounts/index.tsx

```diff
@@ -42,6 +43,48 @@ function Accounts({ className }: Props): React.ReactElement {
     setFilter(filter.toLowerCase());
   }, []);
 
+  const accountsByGenesisHash: { [key: string]: AccountWithChildren[] } = filteredAccount.reduce(
+    (result: { [key: string]: AccountWithChildren[] }, account) => {
+      const { genesisHash } = account;
+      const foundKey = Object.keys(knownGenesis).find((key) => knownGenesis[key].includes(genesisHash ?? ''));
+
+      if (!foundKey) {
+        if (!result.any) {
+          result.any = [];
+        }
+
+        result.any.push(account);
+
+        return result;
+      }
+
+      if (!result[foundKey]) {
+        result[foundKey] = [];
+      }
+
+      const accountWithChildren: AccountWithChildren = { ...account };
+
+      if (account.parentAddress) {
+        const parentAccount = filteredAccount.find(({ address }) => address === account.parentAddress);
+
+        if (parentAccount) {
```

Comment:
To avoid nested ifs
```js
const parentAccount = account.parentAddress &&  filteredAccount.find(({ address }) => address === account.parentAddress);

if (parentAccount) {
```